code/_onclick/hud/new_player.dm

```diff
@@ -11,25 +18,36 @@
 
 	var/list/buttons = subtypesof(/atom/movable/screen/lobby)
 	for(var/button_type in buttons)
-		var/atom/movable/screen/lobby/lobbyscreen = new button_type()
+		var/atom/movable/screen/lobby/lobbyscreen = new button_type(our_hud = src)
 		lobbyscreen.SlowInit()
-		lobbyscreen.hud = src
 		static_inventory += lobbyscreen
+		if(!lobbyscreen.always_shown)
+			toggleable_elements += lobbyscreen
 		if(istype(lobbyscreen, /atom/movable/screen/lobby/button))
 			var/atom/movable/screen/lobby/button/lobby_button = lobbyscreen
 			lobby_button.owner = REF(owner)
 
+/datum/hud/new_player/Destroy()
+	. = ..()
+	QDEL_LIST(toggleable_elements)
+
 /atom/movable/screen/lobby
 	plane = SPLASHSCREEN_PLANE
-	layer = LOBBY_BUTTON_LAYER
+	layer = LOBBY_MENU_LAYER
 	screen_loc = "TOP,CENTER"
+	///Whether this HUD element can be hidden from the client's screen or not
+	var/always_shown = FALSE
+
+/atom/movable/screen/lobby/New(loc, datum/hud/our_hud, ...)
+	if(our_hud)
+		hud = our_hud
```

Comment:
similar, will this hang a hardref?

Reply:
parent handles nulling the hud, if i'm not mistaken

https://github.com/tgstation/tgstation/blob/44c06e1c844886fa931ef7f217f02b1c964a800b/code/_onclick/hud/screen_objects.dm#L39-L42

Reply:
if the hud is deleted and the lobby item is not I mean

Reply:
actually I guess this is fine huh, since we're in an explicit owning relationship. nvm